code/game/gamemodes/devil/game_mode.dm

```diff
@@ -83,6 +53,17 @@
 	text += "	[lawlorify[LORE][ply.devilinfo.banish]]</br></br>"
 	return text
 
+/datum/game_mode/proc/printsoulsowned(datum/mind/devil)
+	if(!owner.current.has_antag_datum(/datum/antag/devil, true))
```

Comment:
Something something wrapper proc
Also you don't need to be calling this twice, here, it can just be the following;

    var/datum/antag/devil/devil_datum = owner.current.has_antag_datum(/datum/antag/devil, true) //but with a wrapper proc!!!
    if(!devil_datum)
    	return